fe1-web/src/features/linked-organizations/components/ManualInputModal.tsx

```diff
@@ -0,0 +1,135 @@
+  const [manualPublicKey, setManualPublicKey] = useState<string>('');
+  const [manualServerAddress, setManualServerAddress] = useState<string>('');
+  const [manualChallengeValue, setManualChallengeValue] = useState<string>('');
+  const [manualChallengeValidUntil, setManualChallengeValidUntil] = useState<Timestamp>(
+    Timestamp.EpochNow().addSeconds(86400),
+  );
+  const [startDate, setStartDate] = useState(manualChallengeValidUntil.toDate());
+
+  return (
+    <Modal testID="modal-manual-input" transparent visible={visible} onRequestClose={onClose}>
+      <TouchableWithoutFeedback containerStyle={ModalStyles.modalBackground} onPress={onClose} />
+      <ScrollView style={ModalStyles.modalContainer}>
+        <ModalHeader onClose={onClose}>
+          {STRINGS.linked_organizations_addmanually_title}
+        </ModalHeader>
+        <Text style={Typography.paragraph}>{STRINGS.linked_organizations_enterLaoId}</Text>
+        <Input
+          testID="modal-input-laoid"
+          value={manualLaoId}
+          onChange={setManualLaoID}
+          placeholder={STRINGS.linked_organizations_placeholderLaoID}
+        />
+        <Text style={Typography.paragraph}>{STRINGS.linked_organizations_enterPublicKey}</Text>
+        <Input
+          testID="modal-input-publickey"
+          value={manualPublicKey}
+          onChange={setManualPublicKey}
+          placeholder={STRINGS.linked_organizations_placeholderPublicKey}
+        />
+        <Text style={Typography.paragraph}>{STRINGS.linked_organizations_enterServerAddress}</Text>
+        <Input
+          testID="modal-input-serveraddress"
+          value={manualServerAddress}
+          onChange={setManualServerAddress}
+          placeholder={STRINGS.linked_organizations_placeholderServerAddress}
+        />
+        <Text style={Typography.paragraph}>{STRINGS.linked_organizations_enterChallengeValue}</Text>
+        <Input
+          testID="modal-input-challengeval"
+          value={manualChallengeValue}
+          onChange={setManualChallengeValue}
+          placeholder={STRINGS.linked_organizations_placeholderChallengeValue}
+        />
+
+        <Text style={Typography.paragraph}>
+          {STRINGS.linked_organizations_enterChallengeValidUntil}
+        </Text>
+        <DatePicker
+          selected={startDate}
+          onChange={(date: Date) => {
+            setStartDate(date);
+            setManualChallengeValidUntil(Timestamp.fromDate(date));
+          }}
+        />
+
+        <PoPTextButton
+          testID="add-manually"
+          onPress={() => {
+            if (
+              (isInitiatingOrganizer &&
+                (!manualLaoId ||
+                  !manualPublicKey ||
+                  !manualServerAddress ||
+                  !manualChallengeValue ||
+                  manualChallengeValidUntil <= Timestamp.EpochNow())) ||
+              (!isInitiatingOrganizer && (!manualLaoId || !manualPublicKey || !manualServerAddress))
```

Comment:
With this simple logic it is okay-ish (I know we don't have proper form validation in other parts of the app) but the error message could be much more useful if it only showed relevant information such as error X with field Y.

Reply:
Maybe we should create an issue for proper form validation..

Reply:
I changed it now so that it provides more precise error messages and I displayed that on the Modal as an error message and not in the toaster, because we still have the problem of the toaster being behind modals. I tried looking into that again but everything I tried did not work unfortunately.

Reply:
I see 😅 Great that you added better error messages though 😁